### ISSUE TITLE
Disabling synchronization on authentication doesn't work

### ISSUE THREAD

**Issue by dottodot** [feathersjs-ecosystem/feathers-sync#106]
I've all of a sudden started getting an error of `app.emit is not a function` when users try to log in. I've not be able to work out exactly why but figured it was something to do with `feathers-sync`. So I've tried to disable it in my authentication hooks but it has no effect.

```
app.service('authentication').hooks({
    before: {
      create: [
        lowerCase('email'),
        authentication.hooks.authenticate(config.strategies),
        customizeJWTPayload()
      ]
    },
    after: {
      create: [
        context => {
          context[SYNC] = false;
          return context;
        }
      ]
    }
  });

**Comment by dottodot** [comment]
In fact I don't think disabling synchronization works at all just tried on another service and `ctx[SYNC]` is always undefined.

```
app.mixins.push((service, path) => {
    service._emit = service.emit;
    service.mixin({
      emit(event, data, ctx) {
        console.log(ctx[SYNC]);

        const disabled = ctx && ctx[SYNC] === false;

        if (!service._serviceEvents.includes(event) || disabled) {
          debug(`Passing through non-service event '${path} ${event}'`);
          return this._emit(event, data, ctx);
        }

        const context = hooks.isHookObject(ctx)
          ? _.omit(ctx, 'app', 'service')
          : ctx;

        debug(`Sending sync-out event '${path} ${event}'`);

        return app.emit('sync-out', {
          event,
          path,
          data,
          context
        });
      }
    });
  });
```

**Comment by hatchett** [comment]
Did you manage to resolve this? I'm having the same issue. Disabling SYNC as per the docs doesn't seem to work at all.

**Comment by dottodot** [comment]
I'm afraid not.

**Comment by hatchett** [comment]
`const disabled = ctx && ctx.sync === false;`

A quick fix: change the line above in feathersjs-sync/core.js and then add a sync boolean to the context in your after hooks. 

`context.sync = false;`

**Comment by daffl** [comment]
The `app.emit` error probably is not related to this module and is tracked in https://github.com/feathersjs/feathers/issues/1183.

**Comment by hatchett** [comment]
I never had the app.emit error, just couldn't disable sync in hooks.

**Comment by daffl** [comment]
Fixed in v1.1.1
